### PR TITLE
[T-85] Authorization backend vol. 2

### PR DESCRIPTION
**Pull request by lakatop**
* Authorization login (obtain token) and refresh token is now implemented via build in jwt functions
* Added custom register and logout (blacklist refresh token) views
* Views are now protected with authorization - registration and logout views are exceptions (they can be accesed even without authorization)
* Testing view for authorization userView

### CODE REVIEW

File: src/backend/core/user/urls.py [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 
 urlpatterns = [
     path("userView", views.UserView.as_view()),
     path("register", views.RegistrationView.as_view()),
-    path("login", views.LoginView.as_view()),
-    path("logout", views.LogoutView.as_view()),
-    path("refreshToken", views.RefreshTokenView.as_view()),
+    path("login", jwt_views.TokenObtainPairView.as_view(), name="token_create"),
+    path("logout/blacklist", views.BlacklistTokenView.as_view()),
+    path("refreshToken", jwt_views.TokenRefreshView.as_view(), name="token_refresh"),
```
Comment (michalpulpan) [inline src/backend/core/user/urls.py:11]: don't you want to keep url path consistency (without camel case, etc)

File: src/backend/core/user/views.py [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
-
-import jwt
 
 
 class RegistrationView(APIView):
     """
     View for registering new users.
     """
 
+    # All views have authentication backend by default
```
Comment (michalpulpan) [inline src/backend/core/user/views.py:17]: How about AllowAny by default?

**Review by michalpulpan — COMMENTED** [review]
please see my suggestions

File: src/backend/core/user/views.py (hunk 2 above)
Comment (lakatop) [reply src/backend/core/user/views.py:17]: AllowAny was deleted to assure authentication

Comment (michalpulpan) [reply src/backend/core/user/views.py:17]: I see okay, thanks

File: src/backend/core/user/urls.py (hunk 1 above)
Comment (michalpulpan) [reply src/backend/core/user/urls.py]: Good boy

**Review by michalpulpan — APPROVED** [review]

**Review by VL-CZ — APPROVED** [review]

**Pull request merged**